ambry-api/src/main/java/com/github/ambry/quota/RejectingQuotaChargeCallback.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.quota;
+
+import com.github.ambry.rest.RestRequest;
+import com.github.ambry.router.RouterErrorCode;
+import com.github.ambry.router.RouterException;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+
+/**
+ * A {@link QuotaChargeCallback} implementation that will reject requests that exceed their quota.
+ */
+public class RejectingQuotaChargeCallback implements QuotaChargeCallback {
```

Comment:
Just wondering if it is okay for this class to be in `ambry-api` module or if it should go in `ambry-quota` module since it is an implementation.

Reply:
Good call. Move around a few files to make that happen.